src/Previewer/utils.js

```diff
@@ -2,13 +2,23 @@ import GObject from "gi://GObject";
 import Gtk from "gi://Gtk";
 import Gio from "gi://Gio";
 import Adw from "gi://Adw";
+import GIRepository from "gi://GIRepository";
+// const repository = GIRepository.Repository.get_default();
 
+const namespaces = getNamespaces();
 export function getObjectClass(class_name) {
-  const split = class_name.split(/(?=[A-Z])/);
-  if (split.length < 2) return;
+  const namespace = namespaces.find((namespace) =>
+    class_name.startsWith(namespace),
+  );
```

Comment:
This might not be quite good enough. I could imagine a case where `Gtk` matches `GtkSourceCompletionProvider` before `GtkSource`. Maybe the should be sorted first?

Reply:
good call didn't think of that

fixed in https://github.com/sonnyp/Workbench/pull/366/commits/9a01524a8e24f25367270be1511bbf3fef78a61d